### PR TITLE
[2862] Get protocol from client config base url

### PR DESCRIPTION
**Pull request by jsanda**
This PR is for [issue 2862](https://github.com/apache/incubator-openwhisk/issues/2862).

### CODE REVIEW

File: tools/cli/go-whisk-cli/commands/api.go [hunk 1]
```diff
@@ -818,7 +818,7 @@ func parseApi(cmd *cobra.Command, args []string) (*whisk.Api, *QualifiedName, er
     } else {
         urlActionPackage = "default"
     }
-    api.Action.BackendUrl = "https://" + Client.Config.Host + "/api/v1/web/" + qName.GetNamespace() + "/" + urlActionPackage + "/" + qName.GetEntity() + ".http"
+    api.Action.BackendUrl = Client.Config.BaseUrl.Scheme + "://" + Client.Config.Host + "/api/v1/web/" + qName.GetNamespace() + "/" + urlActionPackage + "/" + qName.GetEntity() + ".http"
```
Comment (dubeejw) [inline tools/cli/go-whisk-cli/commands/api.go]: Need a test for this.

Comment (jsanda) [reply tools/cli/go-whisk-cli/commands/api.go]: @dubeejw where do CLI tests live?

**Comment by mdeuser** [comment]
Fixes #2862

**Comment by jsanda** [comment]
Thanks. I will start looking at the test code tonight.

**Comment by rabbah** [comment]
fwiw you can test http directly by directly hitting the controller.

**Comment by jsanda** [comment]
@rabbah how can I determine the controller hostname from within tests?

**Comment by rabbah** [comment]
https://github.com/apache/incubator-openwhisk/blob/master/tests/src/test/scala/common/WhiskProperties.java line 229

**Comment by dubeejw** [comment]
@jsanda, as pointed out above, you can use `WhiskProperties.getApiHostForClient()`.

**Comment by jsanda** [comment]
I have been away for the past several days. I will update this PR tomorrow.

**Comment by dubee** [comment]
Closing this one as the CLI has moved to incubator-openwhisk-cli and incubator-openwhisk-client-go.

**Pull request closed without merging**